src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,156 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> first;
+    private Node<T> last;
+    private int size;
+
     @Override
     public void add(T value) {
+        Node<T> lastNode = last;
+        Node<T> newNode = new Node<>(lastNode, value, null);
+        last = newNode;
+        if (first == null) {
+            first = newNode;
+        } else {
+            lastNode.next = newNode;
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        checkForIndexEqualSize(index);
+        if (index == size) {
+            add(value);
+        } else if (index == 0) {
+            Node<T> actual = getNode(index);
+            Node<T> newNode = new Node<>(null, value, actual);
+            actual.prev = newNode;
+            first = newNode;
+            size++;
+        } else {
+            Node<T> actual = getNode(index);
+            Node<T> newNode = new Node<>(actual.prev, value, actual.next);
+            newNode.prev = actual.prev;
+            actual.prev.next = newNode;
+            actual.prev = newNode;
+            newNode.next = actual;
+            size++;
+        }
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (T element : list) {
+            add(element);
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkForIndexBound(index);
+        return getNode(index).value;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        checkForIndexBound(index);
+        Node<T> currentNode = getNode(index);
+        T oldValue = currentNode.value;
+        currentNode.value = value;
+        return oldValue;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        checkForIndexBound(index);
+        Node<T> nodeForRemove = getNode(index);
+        return unlinkNode(nodeForRemove);
     }
 
     @Override
     public boolean remove(T object) {
+        if (object == null) {
+            for (Node<T> current = first; current != null; current = current.next) {
+                if (current.value == null) {
+                    unlinkNode(current);
+                    return true;
+                }
+            }
+        } else {
+            for (Node<T> current = first; current != null; current = current.next) {
+                if ((current.value != null) && (current.value.equals(object))) {
+                    unlinkNode(current);
+                    return true;
+                }
+            }
+        }
         return false;
```

Comment:
as an option, you can use the loop "while", while at the same time take the creation of the "current" out of it. This can make the code easier to read and size.